xarray/core/dtypes.py

```diff
@@ -20,9 +20,9 @@ def maybe_promote(dtype):
     fill_value : Valid missing value for the promoted dtype.
     """
     # N.B. these casting rules should match pandas
-    if np.issubdtype(dtype, float):
+    if np.issubdtype(dtype, np.floating):
```

Comment:
I was getting these warnings - lmk if this is the right change
> FutureWarning: Conversion of the second argument of issubdtype from `complex` to `np.complexfloating` is deprecated. In future, it will be treated as `np.complex128 == np.dtype(complex).type`.
    elif np.issubdtype(dtype, complex):